### PR TITLE
Remove snapshot command

### LINKED ISSUE #1690: Remove `snapshot` command in client CLI
## Why
We need to remove the deprecated `snapshot` command of the client CLI. It will be released with the next distribution (`2423`).

## What
Remove the command and warn our users that the command will be removed so that they can use the new `cardano-db` command instead.

## How

- [x] Remove the `snapshot` command in the client CLI
- [x] Add an error message when `snapshot` command is used or when a command is not found (but `snapshot` is not displayed anymore in the help)
- [x] Remove the `snapshot` command in the documentation
- [x] Update the minor version of the client CLI as it is a breaking change
- [ ] Warn in advance all users with an announcement in slack/discord

## Later
- Create a blog post when distribution is released

### PR DESCRIPTION
**Pull request by sfauvel**
## Content
This PR remove `snapshot`command from Mithril client cli.

Initial behavior:
image

New behavior:
image

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [ ] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [x] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)
Closes #1690

### CODE REVIEW

File: mithril-client-cli/src/main.rs [hunk 1]
```diff
@@ -139,6 +139,27 @@ impl Args {
 
         Ok(Logger::root(Arc::new(drain), slog::o!()))
     }
+
+    fn parse_with_decorator(
+        decorator: &dyn Fn(Result<Self, clap::error::Error>) -> Result<Self, clap::error::Error>,
```
Comment (jpraynaud) [inline mithril-client-cli/src/main.rs:144]: Can we use a `StdResult` instead?
Also can we alias `clap::error::Error` with `ClapError`?

File: mithril-client-cli/src/main.rs [hunk 2]
```diff
@@ -139,6 +139,27 @@ impl Args {
...
+    }
+
+    fn handle_deprecated_decorator(
+        args_result: Result<Self, clap::error::Error>,
+    ) -> Result<Self, clap::error::Error> {
+        let styles = Args::command().get_styles().clone();
+        Deprecation::handle_deprecated_commands(
+            args_result,
+            styles,
+            vec![DeprecatedCommand::new("snapshot", "cardano-db")],
```
Comment (jpraynaud) [inline mithril-client-cli/src/main.rs:160]: This could be part of the configuration of the `parse_with_decorator` function IMO.

**Review by jpraynaud — APPROVED** [review]
LGTM :+1: 

I left few comments below.

**Review by Alenar — APPROVED** [review]

File: mithril-client-cli/src/main.rs (hunk 2 above)
Comment (sfauvel) [reply mithril-client-cli/src/main.rs]: Done

File: mithril-client-cli/src/main.rs (hunk 1 above)
Comment (sfauvel) [reply mithril-client-cli/src/main.rs]: We have made the ClapError alias.
We keep clap Error for the decorator to have the same signature of the decorated function (try_parse).

**Pull request merged**